### PR TITLE
Scrapy with Tor

### PR DESCRIPTION
**Pull request by MohammadShahin**
Scrapy with Tor

### CODE REVIEW

File: task3/scrapy_tor/scrapy_tor/middlewares.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from stem import Signal
+from stem.control import Controller
+from scrapy.downloadermiddlewares.httpproxy import HttpProxyMiddleware
+import os
+from dotenv import load_dotenv
+
+
+load_dotenv()
```
Comment (ilov3) [inline task3/scrapy_tor/scrapy_tor/middlewares.py:8]: Not a good place to load project configuration env variables. I would prefer to see it in some init/bootstrap step.

File: task3/scrapy_tor/scrapy_tor/middlewares.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+N = int(os.getenv('ALLOWED_REQUESTS'))
```
Comment (ilov3) [inline task3/scrapy_tor/scrapy_tor/middlewares.py:9]: This should be configurable through `settings.py` module

File: task3/scrapy_tor/scrapy_tor/middlewares.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    def process_request(self, request, spider):
+        """
+        This method is used to change the TOR identity every N requests.
+        """
+        # if ProxyMiddleware.num_requests == N:
+        #     new_tor_identity()
+        #     ProxyMiddleware.num_requests = 1
+        # else:
+        #     ProxyMiddleware.num_requests += 1
+        new_tor_identity()
```
Comment (ilov3) [inline task3/scrapy_tor/scrapy_tor/middlewares.py:43]: Where is the required logic of changing identity per `N` requests?

Comment (MohammadShahin) [reply task3/scrapy_tor/scrapy_tor/middlewares.py:43]: The lines in comments are used for that. I was just debugging, and I forgot about them. Please check again.

**Pull request merged**